tracing-core/src/dispatch.rs

```diff
@@ -169,6 +168,18 @@ pub struct Dispatch {
     collector: &'static (dyn Collect + Send + Sync),
 }
 
+/// `WeakDispatch` is a version of [`Dispatch`] that holds a non-owning reference
+/// to a collector. This collector is accessed calling [`WeakDispatch::upgrade`],
+/// which returns an `Option<Dispatch>`.
```

Comment:
it might be worth summarizing why this type is used (maybe just by linking to the "Avoiding Memory Leaks" section in the `Collect` docs)? also, we should maybe mention `std::sync::Weak` here.